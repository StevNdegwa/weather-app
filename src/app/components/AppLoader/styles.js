import { styled, Box } from "@mui/material";

export const AppLoaderContent = styled(Box)`
width: 100%;
height: 100%;
display: flex;
justify-content: center;
align-items: center;
font-size:1.3rem;
font-weight:800;
color: darkgray;
background-color:transparent;
`;

export const WeatherIconContainer = styled(Box)`
position: absolute;
top:0;
right:0;
margin:10px;
`;