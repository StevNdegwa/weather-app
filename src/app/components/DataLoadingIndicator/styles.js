import { Box, styled } from "@mui/material";

export const LoadingIndicatorWrapper = styled(Box)`
position: absolute;
top: 0;
left: 0;
z-index: var(--depth-overlay);
width: 100%;
height: 100%;
display: flex;
justify-content: center;
align-items: center;
background-color: hsla(0, 0%, 100%, 0.2);
backdrop-filter: blur(3px);
`;
