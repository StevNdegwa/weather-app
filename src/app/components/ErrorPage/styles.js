import { styled, Box } from "@mui/material";

export const ErrorPageWrapper = styled(Box)`
width: 100%;
height: 100%;
display: flex;
justify-content: center;
align-items: center;
`;

export const ErrorPageContent = styled(Box)`
text-align:center;
font-size: var(--font-size-md);
`;
