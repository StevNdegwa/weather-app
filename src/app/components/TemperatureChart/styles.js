import { Stack, styled } from "@mui/material";

export const ChartWrapper = styled(Stack)`
max-width: 700px;
height: 400px;
background-color: white;
position: relative;
border-radius: var(--border-radius-md);
margin: 5rem auto;
overflow: auto;
& > div{
  width: 700px;
  height:100%;
  padding: 0.5rem;
}
@media only screen and (max-width: 800px){
  margin: 0 auto;
  border-radius: 0;
}
& svg{
  font-size: 0.8rem;
  & path{
    fill: #90CBEC;
    cursor: pointer;
  }
}
`;