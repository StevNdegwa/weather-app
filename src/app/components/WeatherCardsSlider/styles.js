import { styled, Slider } from "@mui/material";
import { Slide } from "react-slideshow-image";

export const WeatheCardsSliderWrapper = styled(Slide)`
width: 95%;
background-color: none;
overflow: hidden;
border-radius: var(--border-radius-lg);
& button:disabled{
  opacity: 0;
}
`;

export const IndexSlider = styled(Slider)`
pointer-events: none;
& .MuiSlider-track {
  border: none;
  background-color: #90CBEC !important;
}
& .MuiSlider-rail {
  border: none;
  background-color: #DAE3E8 !important;
}
& .MuiSlider-thumb{
  display: none;
}
`;
