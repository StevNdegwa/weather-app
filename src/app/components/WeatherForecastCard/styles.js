import { Card, styled, Typography } from "@mui/material";

export const CardWrapper = styled(Card)`
width: 200px;
height: 180px;
margin: auto;
background-color: hsla(0, 0%, 100%, 0.2);
backdrop-filter: blur(20px);
border-radius: var(--border-radius-lg); 
border: 1px outset transparent;
box-shadow: none;
cursor: pointer;
display: flex;
justify-content:center;
align-items:center;
${({ selected }) => selected && "border: 3px solid blue"};
`;

export const Temperature = styled(Typography)`
width: 100%;
height: 2.5rem;
text-align: center;
font-weight: bold;
font-size: var(--font-size-md);
`;

export const Text = styled(Typography)`
width: 100%;
height: 1rem;
text-align: center;
font-size: var(--font-size-sm);
`;
