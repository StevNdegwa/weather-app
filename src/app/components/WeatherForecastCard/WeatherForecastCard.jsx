import { useCallback, useMemo } from "react";
import { CardContent } from "@mui/material";
import PropTypes from "prop-types";
import { useSelector, useDispatch } from "react-redux";
import { timeFormat } from "d3-time-format";
import getTemperature from "../../utils/getTemperature";
import { selectDate } from "../../features/weather/selectedDateSlice";
import { CardWrapper, Temperature, Text } from "./styles";

export const formatDate = timeFormat("%a, %b %d");
export const formatTime = timeFormat("%H: %M");

export default function WeatherForecastCard({
  date,
  temperature,
  showTempChart,
}) {
  const { day } = useMemo(() => {
    let d = new Date(date);
    return {
      time: formatTime(d),
      day: formatDate(d),
      date: d,
    };
  }, [date]);

  const selectedScale = useSelector((state) => state.selectedScale);
  const dispatch = useDispatch();

  const selectThisCard = useCallback(() => {
    dispatch(selectDate(date.toJSON()));
    showTempChart();
  }, [date, dispatch, showTempChart]);

  const dispTemperature = useMemo(
    () => getTemperature(temperature, selectedScale),
    [selectedScale, temperature]
  );

  return (
    <CardWrapper onClick={selectThisCard}>
      <CardContent>
        <Text>Avg. temperature</Text>
        <Temperature>{dispTemperature.formattedTemp}</Temperature>
        <Text>{day}</Text>
      </CardContent>
    </CardWrapper>
  );
}

WeatherForecastCard.proTypes = {
  date: PropTypes.instanceOf(Date),
  temperature: PropTypes.number,
  showTempChart: PropTypes.func.isRequired,
};
