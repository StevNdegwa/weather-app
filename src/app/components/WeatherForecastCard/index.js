import WeatherForecastCard from "./WeatherForecastCard";
export default WeatherForecastCard;