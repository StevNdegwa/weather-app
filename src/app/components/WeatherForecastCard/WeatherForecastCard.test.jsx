import { mount } from "enzyme";
import { Provider } from "react-redux";
import store from "../../features/store";
import WeatherForecastCard from "./WeatherForecastCard";
import { CardWrapper, Temperature } from "./styles";
import { act } from "@testing-library/react";
import { formatDate } from "./WeatherForecastCard";

describe("test <WeatherForecastCard/> component", () => {
  let wrapper,
    showTempChart = jest.fn(),
    date = new Date("2021-12-12T07:29:45.123Z"),
    props = {
      date,
      temperature: 120.4,
    };

  beforeEach(
    () =>
      (wrapper = mount(
        <Provider store={store}>
          <WeatherForecastCard {...props} showTempChart={showTempChart} />
        </Provider>
      ))
  );

  test("snapshot testing", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should include all components", () => {
    expect(wrapper.find(CardWrapper)).toHaveLength(1);
    expect(wrapper.find(Temperature)).toHaveLength(1);
  });

  it("should contain weather forecast day", () => {
    expect(wrapper.find(WeatherForecastCard).text()).toEqual(
      expect.stringContaining(formatDate(props.date))
    );
  });

  test("clicking a card should update the selected card state", async () => {
    expect(store.getState().selectedDate).toEqual(null);

    await act(async () => {
      wrapper.find(CardWrapper).prop("onClick")();
    });

    expect(store.getState().selectedDate).toEqual(date.toJSON());
  });
});
