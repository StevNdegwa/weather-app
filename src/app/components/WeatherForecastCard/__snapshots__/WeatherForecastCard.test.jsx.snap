// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <WeatherForecastCard/> component snapshot testing 1`] = `"<div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j94aqb-MuiPaper-root-MuiCard-root\\"><div class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-p09gn3-MuiTypography-root\\">Avg. temperature</p><p class=\\"MuiTypography-root MuiTypography-body1 css-a4lwg8-MuiTypography-root\\"><span>49.11Â°C</span></p><p class=\\"MuiTypography-root MuiTypography-body1 css-p09gn3-MuiTypography-root\\">Sun, Dec 12 2021</p></div></div>"`;
